### PR TITLE
TFTRT: Complete IActivationLayer coverage for TRT 5.1

### PR DESCRIPTION
**Pull request by trevor-m**
* The following ops are now supported for TRT >= 5.1.2: ClipByValue, Elu, Selu, Softsign, Softplus
* Relu6 and LeakyRelu will use the new accelerated IActivationLayer implementations if TRT 5.1.2+ is present
* ConvertActivation now uses `ActivationTypeMap()` similar to how ConvertUnary does.
* Unit tests for all new ops

### CODE REVIEW

**Comment by gbaned** [comment]
@trevor-m could you please resolve the conflicts?

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 1]
```diff
@@ -2505,14 +2520,33 @@ TEST_F(OpConverterTest, ConvertActivation) {
...
+  for (auto& pair : *map) {
+    ops_to_test.push_back(pair.first);
+  }
+  // Add other activation ops to test.
+  ops_to_test.push_back("Relu6");
+  ops_to_test.push_back("LeakyRelu");
   // Ok.
-  for (const string& op_name :
-       {"LeakyRelu", "Relu", "Relu6", "Sigmoid", "Tanh"}) {
+  for (string op_name : ops_to_test) {
```
Comment (aaroey) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc]: nit: `const string&`

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 2]
```diff
@@ -4910,6 +4949,125 @@ TEST_F(OpConverterTest, ConvertArgMinMax) {
...
+    AddTestTensor("clip_value_max", {1});
+    RunValidationAndConversion(node_def, error::UNIMPLEMENTED,
+                               "The input \"clip_value_max\" for ClipByValue "
+                               "must be a constant, at my_clip");
+  }
+
+  TestConvertClipByValue<DT_FLOAT>(this);
+  TestConvertClipByValue<DT_HALF>(this);
+  // TRT does not support INT32 for this layer.
+  // TestConvertClipByValue<DT_INT32>(this);
```
Comment (aaroey) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc]: Does TRT plan to support that? If not just remove this line.

**Review by aaroey — CHANGES_REQUESTED** [review]

**Review by aaroey — APPROVED** [review]
Thanks for the fix!

**Pull request merged**